plugins/chat/app/services/chat/lookup_user_threads.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+module Chat
+  # Gets a list of threads for a user.
+  #
+  # Only threads that the user is a member of with a notification level
+  # of normal or tracking will be returned.
+  #
+  # @example
+  #  Chat::LookupUserThreads.call(guardian: guardian, limit: 5, offset: 2)
+  #
+  class LookupUserThreads
+    include Service::Base
+
+    THREADS_LIMIT = 10
+
+    # @!method call(guardian:, limit: nil, offset: nil)
+    #   @param [Guardian] guardian
+    #   @param [Integer] limit
+    #   @param [Integer] offset
+    #   @return [Service::Base::Context]
+
+    contract
+    step :set_limit
+    step :set_offset
+    model :threads
+    step :fetch_tracking
+    step :fetch_memberships
+    step :fetch_participants
+    step :build_load_more_url
+
+    # @!visibility private
+    class Contract
+      attribute :limit, :integer
+      attribute :offset, :integer
+    end
+
+    private
+
+    def set_limit(contract:, **)
+      context.limit = (contract.limit || THREADS_LIMIT).to_i.clamp(1, THREADS_LIMIT)
+    end
+
+    def set_offset(contract:, **)
+      context.offset = [contract.offset || 0, 0].max
+    end
+
+    def fetch_threads(guardian:, **)
+      ::Chat::Thread
```

Comment:
Most of this query and the one in `LookupChannelThreads` is so similar...it's only 2 places ATM so not too bad, but it could become hard to keep track of all the `where` clauses and includes in both places.

Maybe a new `ThreadListQuery` class could be used in both places, that:

1. Does all this top part of the query with all the `includes` needed for a thread list
2. Includes the common `where` clauses like these:

```
.where("original_message.chat_channel_id = chat_threads.channel_id")
.where("original_message.deleted_at IS NULL")
.where("last_message.chat_channel_id = chat_threads.channel_id")
.where("last_message.deleted_at IS NULL")
.where("chat_threads.replies_count > 0")
.where("user_chat_thread_memberships.user_id = ?", guardian.user.id)
```

Not super necessary to do this right now, but it's another thing to have to think of in future when changing these queries.

Reply:
Yes I hesitated qnd should probably do it in near future I agree